1/ToDoApp/src/backend/resolvers/todo/todos.spec.js

```diff
@@ -0,0 +1,168 @@
+const { mergeResolvers } = require("merge-graphql-schemas");
+const { createTestClient } = require('apollo-server-testing');
+const { ApolloServer, gql } = require('apollo-server');
+const { typeDefs } = require("../../schema/typeDefs");
+const { userResolver } = require("../user/userResolver");
+const { todoResolver } = require("../todo/todoResolver");
+
+const resolvers = mergeResolvers([userResolver, todoResolver]);
+
+const server = new ApolloServer({typeDefs, resolvers});
+const { query, mutate } = createTestClient(server);
+
+describe('query', () => {
+    describe('todos', () => {
+        it('todo_list has 2 initial todos', async () => {
+            const res = await query({
+                query: GET_ALL_TODOS
+            });
+            expect(res.data.allTodos).toHaveLength(2);
+        });
+
+        it('returns todo-message for given id', async () => {
+            const res = await query({
+                query: GET_FIRST_TODOMESSAGE
+            });
+            expect(res.data.todoById).toMatchObject({message: "first todo"});
+        });
+    });
+});
+
+describe('mutate', () => {
+    var logInToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZW1haWwiOiJ5b3VyQGVtYWlsLmNvbSIsImlhdCI6MTU3NTQ0NjYzNCwiZXhwIjoxNjA3MDA0MjM0fQ.cHeP2Ih8Dxeyyxw3rePvfVCeJJFFnO7A9BrV1QL4hcA";
+    describe('given user is not logged in/no token', () => {
+        it('creating a todo returns an error', async () => {
+            const res = await mutate({
+                mutation: CREATE_TODO,
+                variables: {
+                    message: "neues Todo",
+                    token: ""
+                }
+            });
+            expect(res.errors).toHaveLength(1);
+        });
+        it('logIn with correct credentials returns token-String', async () => {
+            const res = await mutate({
+                mutation: LOGIN,
+                variables: {
+                    email: "your@email.com",
+                    password: "password"
+                }
+            });
+            expect(res.data.login).toBeDefined();
+        });
+    });
+    describe('give user is loggedIn', () => {
+        it('adds a todo', async () => {
+            const res_allTodos=await query ({
+                    query: GET_ALL_TODOS
+            });
+            const res = await mutate({
+                mutation: CREATE_TODO,
+                variables: {
+                    message: "neues Todo",
+                    token: logInToken
+                }
+            });
+            expect(res.data.addTodo).toHaveLength(res_allTodos.data.allTodos.length+1);
+            expect(res.data.addTodo[res.data.addTodo.length-1].message).toEqual("neues Todo");
+
+        });
+        describe('Modifying Todos', () => {
+            describe('given the loggedIn-User is the creator of the todo (allowed to modify)', () => {
+
```

Comment:
I would take the opportunity here to put the implementation of `given the loggedIn-User is the creator of the todo (allowed to modify)` into a `beforeEach`:
```js
let variables
beforeEach(() => {
  createTodo({
    id: 1,
    message: "Previous message",
    finished: false,
    creatorId: 4711
  })
  const loginToken = generateTokenFor({userId: 4711})

  variables = {
    ...variables,
    token: logInToken,
    id: 1,
  }
})

```

In your mutation in the test case you're updating todo with id 1 which you created :point_up: as owned by user with id 4711.